### ISSUE TITLE
CMake not properly configuring stepmania

### ISSUE THREAD

**Issue by RedEmerald98** [stepmania/stepmania#1465]
When I'm trying to Configure with CMake, in the log I can see, that a couple files were not found and checking size of some files failed.
Trying to generate then reults in _Error in generation process, project files may be invalid._ ...
Can anybody help me? I have the latest stuff needed installed
Below I uploaded the codes I got, from trying to configure and then trying to generate.
Can anybody help me?
I want to try to implement a code, but I can't do anything to the debug menu without having a proper solution for VS.

[CMakeError-Configuring.txt](https://github.com/stepmania/stepmania/files/1036642/CMakeError-Configuring.txt)
[CMakeError-Generating.txt](https://github.com/stepmania/stepmania/files/1036641/CMakeError-Generating.txt)

Thanks to everyone trying to help me with this problem!

**Comment by chrispable** [comment]
ummm did you init and update the submodules according the instructions?

**Comment by RedEmerald98** [comment]
yes I did. I even downloaded from git with git bash again to try it again to make sure I didn't forget it
still the same errors there...

**Comment by chrispable** [comment]
What directory are you running the commands from? I full console log would help from checkout to failure.

**Comment by chrispable** [comment]
And the git checkout. I want to see every single step your doing. Worst case you may need vs2015 or vs2013 instead of 2017. I don't have 2017 to test.

**Comment by RedEmerald98** [comment]
alright
first the git bash log:

`user@RedEmerald98 MINGW64 /b
$ git clone https://github.com/stepmania/stepmania.git
Cloning into 'stepmania'...
remote: Counting objects: 267751, done.
remote: Compressing objects: 100% (29/29), done.
Receiving objects:  25% (68777/267751), 287.93 MiB | 2.76 MiB/s

user@RedEmerald98 MINGW64 /b
$ git clone https://github.com/stepmania/stepmania.git
Cloning into 'stepmania'...
remote: Counting objects: 267751, done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 267751 (delta 5), reused 9 (delta 5), pack-reused 267717
Receiving objects: 100% (267751/267751), 494.00 MiB | 3.37 MiB/s, done.
Resolving deltas: 100% (179727/179727), done.
Checking out files: 100% (4698/4698), done.

user@RedEmerald98 MINGW64 /b
$ cd stepmania/

user@RedEmerald98 MINGW64 /b/stepmania (master)
$ git submodule update --init
Submodule 'extern/cppformat' (https://github.com/stepmania/fmt.git) registered for path 'extern/cppformat'
Submodule 'extern/ffmpeg-git' (https://github.com/stepmania/ffmpeg.git) registered for path 'extern/ffmpeg-git'
Submodule 'extern/googletest' (https://github.com/stepmania/googletest.git) registered for path 'extern/googletest'
Submodule 'extern/libogg-git' (https://github.com/stepmania/ogg.git) registered for path 'extern/libogg-git'
Submodule 'extern/libvorbis-git' (https://github.com/stepmania/vorbis.git) registered for path 'extern/libvorbis-git'
Submodule 'extern/tomcrypt' (https://github.com/stepmania/libtomcrypt.git) registered for path 'extern/tomcrypt'
Submodule 'extern/tommath' (https://github.com/stepmania/libtommath.git) registered for path 'extern/tommath'
Cloning into 'B:/stepmania/extern/cppformat'...
Cloning into 'B:/stepmania/extern/ffmpeg-git'...
Cloning into 'B:/stepmania/extern/googletest'...
Cloning into 'B:/stepmania/extern/libogg-git'...
Cloning into 'B:/stepmania/extern/libvorbis-git'...
Cloning into 'B:/stepmania/extern/tomcrypt'...
Cloning into 'B:/stepmania/extern/tommath'...
Submodule path 'extern/cppformat': checked out '0b097da31eae1642dd8132f2996e74f7fe354e46'
Submodule path 'extern/ffmpeg-git': checked out 'eda6effcabcf9c238e4635eb058d72371336e09b'
Submodule path 'extern/googletest': checked out '0dd5ef8bd4b9e3978955e8cdab2eec7016dda4c2'
Submodule path 'extern/libogg-git': checked out '18c401c6bc8814d06f3ae53ebf5d4399f90871cc'
Submodule path 'extern/libvorbis-git': checked out 'd8ffc480787fcd8b5bb7203d6e0acf3bbfb2dd02'
Submodule path 'extern/tomcrypt': checked out 'e24b01d3925265c4999ad9a63f3d8707af201952'
Submodule path 'extern/tommath': checked out '3aba4eacc639dc85b3bc93bf565d2aea91e1ecb7'
`
now the Cmake log:
`The C compiler identification is MSVC 19.10.25019.0
The CXX compiler identification is MSVC 19.10.25019.0
Check for working C compiler: E:/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe
Check for working C compiler: E:/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe -- works
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Check for working CXX compiler: E:/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe
Check for working CXX compiler: E:/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.10.25017/bin/HostX86/x86/cl.exe -- works
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Detecting CXX compile features
Detecting CXX compile features - done
Looking for include file alloca.h
Looking for include file alloca.h - not found
Looking for include file assert.h
Looking for include file assert.h - found
Looking for include file dlfcn.h
Looking for include file dlfcn.h - not found
Looking for include file dirent.h
Looking for include file dirent.h - not found
Looking for include file errno.h
Looking for include file errno.h - found
Looking for include file fcntl.h
Looking for include file fcntl.h - found
Looking for include file inttypes.h
Looking for include file inttypes.h - found
Looking for include file math.h
Looking for include file math.h - found
Looking for include file memory.h
Looking for include file memory.h - found
Looking for include file stdarg.h
Looking for include file stdarg.h - found
Looking for include file stddef.h
Looking for include file stddef.h - found
Looking for include file stdint.h
Looking for include file stdint.h - found
Looking for include file stdlib.h
Looking for include file stdlib.h - found
Looking for include file strings.h
Looking for include file strings.h - not found
Looking for include file string.h
Looking for include file string.h - found
Looking for include file unistd.h
Looking for include file unistd.h - not found
Looking for include file sys/param.h
Looking for include file sys/param.h - not found
Looking for include file sys/stat.h
Looking for include file sys/stat.h - found
Looking for include file sys/types.h
Looking for include file sys/types.h - found
Looking for include file sys/utsname.h
Looking for include file sys/utsname.h - not found
Looking for include file sys/wait.h
Looking for include file sys/wait.h - not found
Looking for include file endian.h
Looking for include file endian.h - not found
Looking for include file sys/endian.h
Looking for include file sys/endian.h - not found
Looking for include file machine/endian.h
Looking for include file machine/endian.h - not found
Looking for _mkdir
Looking for _mkdir - found
Looking for _snprintf
Looking for _snprintf - found
Looking for fcntl
Looking for fcntl - not found
Looking for fork
Looking for fork - not found
Looking for mkdir
Looking for mkdir - found
Looking for snprintf
Looking for snprintf - found
Looking for waitpid
Looking for waitpid - not found
Looking for strtof
Looking for strtof - found
Looking for size_t
Looking for size_t - not found
Looking for size_t
Looking for size_t - not found
Looking for size_t
Looking for size_t - not found
Looking for posix_fadvise
Looking for posix_fadvise - not found
Check size of int16_t
Check size of int16_t - done
Check size of uint16_t
Check size of uint16_t - done
Check size of u_int16_t
Check size of u_int16_t - failed
Check size of int32_t
Check size of int32_t - done
Check size of uint32_t
Check size of uint32_t - done
Check size of u_int32_t
Check size of u_int32_t - failed
Check size of int64_t
Check size of int64_t - done
Check size of char
Check size of char - done
Check size of unsigned char
Check size of unsigned char - done
Check size of short
Check size of short - done
Check size of unsigned short
Check size of unsigned short - done
Check size of int
Check size of int - done
Check size of unsigned int
Check size of unsigned int - done
Check size of long
Check size of long - done
Check size of unsigned long
Check size of unsigned long - done
Check size of long long
Check size of long long - done
Check size of float
Check size of float - done
Check size of double
Check size of double - done
Check size of intptr_t
Check size of intptr_t - done
Check size of pid_t
Check size of pid_t - failed
Check size of size_t
Check size of size_t - done
Check size of ssize_t
Check size of ssize_t - failed
Check if the system is big endian
Searching 16 bit integer
Check size of unsigned short
Check size of unsigned short - done
Using unsigned short
Check if the system is big endian - little endian
Checking for function prototype capabilities
Checking for function prototype capabilities - found
Checking for external name shortening requirements
Checking for external name shortening requirements - not needed
Checking if incomplete types are broken.
Checking if incomplete types are broken. - not broken
Could NOT find nasm (missing:  NASM_EXECUTABLE) 
Could NOT find yasm (missing:  YASM_EXECUTABLE) 
Could NOT find BZip2 (missing:  BZIP2_LIBRARIES BZIP2_INCLUDE_DIR) 
Could NOT find ICONV (missing:  ICONV_LIBRARIES ICONV_INCLUDE_DIR) 
Looking for pthread.h
Looking for pthread.h - not found
Found Threads: TRUE  
Looking for pthread_mutex_timedlock
Looking for pthread_mutex_timedlock - not found
Looking for pthread_cond_timedwait
Looking for pthread_cond_timedwait - not found
Found DIRECTX: C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)/Include  
Configuring done`

The Directory I'm using is B: RAMDisk (last time I already tried my E: Data drive and C:).
Also I have installed vs 15 2017. Might VS be the problem?

**Comment by chrispable** [comment]
Msvc2015 might be why but a few things.

Why are you cloning twice? Your submodule syntax is not exactly as stated in the readme:

Before you you can use CMake, make sure the git submodules are initialized. In the parent directory, run git submodule init and git submodule update to have the submodules initialized.


You provided cmake output but not the exact command sequence to get there. Leave NOTHING unsaid.

**Comment by RedEmerald98** [comment]
oh sorry, I forgot to write that. I cloned twice, because I lost the internet connection. To be on the safe route, I deleted the stepmania folder and then I started to clone again

can you please tell me what I exactly have to do so that I would be on the successful route? I can't find an actual tutorial of how to install it :/

**Comment by chrispable** [comment]
I'm merely trying to assist. I literally followed their directions word for word. I have installed the following msvc editions: 2010,2013,2015. Beyond that unless I know what was done keystroke for keystroke I can't offer any other help. I copy and pasted the submodule commands from the readme in my reply above, I guess try that. I'm not familiar if that's any different to what you did or not

**Comment by RedEmerald98** [comment]
that's the order of the steps I did, hope that'll help:
1. `git clone https://github.com/stepmania/stepmania.git`
2. `cd stepmania/`
3. `git submodule update --init`
4. Open CMake (cmake-gui in this case)
5. Set both the source code and binary build location to where the repo was cloned into (in my case B:/stepmania)
6. Press Configure (my Generator was set to VS 15 2017, the blank one, not Win64 or ARM)
**and that's the point where I'm stuck now**
7. Click Generate and then Open Project
8. Set Solution Configuration to "Release"
9. Go to the Build menu and press "Build Solution" (building might take a while...)

is that right?

**Comment by chrispable** [comment]
OK that might be the disconnect, I use cmake command line. Only thing prior to that, that differs is the submodule commands.  If changing that doesn't fix your issue, try vs2015.

**Comment by RedEmerald98** [comment]
changing what and how?
this git project is the first one I'm trying to build so I don't know much about git and cmake

**Comment by chrispable** [comment]
Once again your submodule command does not match exactly. I will paste the readme again:

Before you you can use CMake, make sure the git submodules are initialized. In the parent directory, run git submodule init and git submodule update to have the submodules initialized.

**Comment by RedEmerald98** [comment]
alright, I'll try again from zero with these two steps

**Comment by RedEmerald98** [comment]
after doing `git submodule init` and `git submodule update` do I continue with Cmake?

**Comment by chrispable** [comment]
Yes. I've not used the GUI but from the stepmania folder is where I run it on the command line successfully every time. Not inside source or anything.

**Comment by RedEmerald98** [comment]
still didn't work
installing VS 2014

**Comment by RedEmerald98** [comment]
Update: Bad News!
Configuring still got me the same errors, even though I used VS 14 2015.
```
Could NOT find nasm (missing:  NASM_EXECUTABLE) 
Could NOT find yasm (missing:  YASM_EXECUTABLE) 
Could NOT find BZip2 (missing:  BZIP2_LIBRARIES BZIP2_INCLUDE_DIR) 
Could NOT find ICONV (missing:  ICONV_LIBRARIES ICONV_INCLUDE_DIR) 
Configuring done
Generating done
```
^ this is what it said after I pressed `Configure` again

**Comment by chrispable** [comment]
You didn't finish installing it. You forgot to install the C++ components. Start a c++ project with it and it should download them
